action_server_bt/src/action_server_bt.cpp

```diff
@@ -41,9 +41,13 @@ ActionServerBT::ActionServerBT(const rclcpp::NodeOptions& options, const UserCal
 
   // register the users Plugins and BehaviorTree.xml files into the factory
   register_behavior_trees(params_, factory_, node_);
+  registerNodesIntoFactory(factory_);
+
+  // No one "own" this blackboard
```

Comment:
we want this to persist, by default

Reply:
How would a user go about clearing the global blackboard? Seems like some runs they could want values to persist but others should start clean. Should we consider adding an option to the Action goal?

Reply:
I think this also leads into the feature of the user waning to clear and `ImportBlackboardFromJSON`. Another nice addition to the Action goal would be the ability to specify a path to a JSON file that should be loaded before running the tree.

Reply:
you can still access it and clean it in `onTreeCreated`.

I would use that callback to do all this initialization and let the user take care of it, instead of tryingto figure out all the potential things they may want to do

Reply:
So I was thinking about that as an option too but it could lead to a few situations. Let's say they offer a service to set the blackboard. What would stop them from mutating the BB while the tree is rubbing? Does BT.CPP have guards to protect from reading and writing BB variables at the same time?